### PR TITLE
language: compile all documented example forms

### PR DESCRIPTION
**Pull request by hhenson**
## Summary

- compile every checked-in HGL example through the generated C++ package path
- lower nominal and generic structs, structural construction and sparse deltas, generic operators, and fixed or duration windows
- lower concise graph functions plus runtime collection iteration and predicates, logger injection, sinks, prior output, and keyed TSD writes
- add public static-schema support needed to preserve nominal hierarchy, generic identity, and concrete duration-window metadata
- add generated behavior tests and update the user, developer, architecture, and roadmap documentation to match the implemented boundary

## Stack

- depends on #664, which restores the pre-v0.2 `hg_cpp` formatting policy, makes generated C++ formatter-clean, and replaces generated operator subclasses with transparent contract aliases
- this PR contains one implementation commit on top of that base

## Validation

- all 24 language CTest cases pass, including native compilation of all six examples
- generated headers and sources pass clang-format dry-run with warnings as errors and contain no classes derived from `hgraph::Operator`
- fresh macOS native build and independent Linux validation: 1,718/1,718 native tests pass on each platform
- stable-ABI wheels built with Python 3.12 and installed into fresh Python 3.14 environments on macOS and Linux: 3,123 passed, 10 skipped on each platform
- installed wheel SDK consumer configured, built, and passed using public headers

### CODE REVIEW

File: language/src/codegen/cpp_emitter.cpp [hunk 1]
```diff
@@ -1718,7 +2346,99 @@ namespace hgl::codegen
...
+                            {
+                                value.code = "static_cast<hgraph::Int>(" + raw + ")";
+                            }
+                            else if (map_key)
+                            {
+                                value.code = raw + ".checked_as<" + value_type(type, stmt.range) + ">()";
+                            }
+                            else
+                            {
+                                value.code = raw;
```
[the remark this answers is not in the document]
Comment (hhenson) [reply language/src/codegen/cpp_emitter.cpp]: Verified against the current typed TSS API: In<Name, TSS<T>>::added(), removed(), and values() return std::vector<T>, so this range element is Str rather than ValueView. The generated collection fixture compiles in the complete native build; no conversion is required here.

File: include/hgraph/types/type_pattern.h [hunk 2]
```diff
@@ -605,6 +627,17 @@ namespace hgraph
...
     };
 
+    template <typename ValueBundle, typename... Fields> struct ts_pattern_lower<NominalTSB<ValueBundle, Fields...>>
+    {
+        [[nodiscard]] static TypePattern lower()
+        {
+            const auto *value = value_schema_descriptor<ValueBundle>::value_meta();
+            return TypePattern::tsb(type_pattern_detail::tsb_field_names<Fields...>(),
+                                    type_pattern_detail::tsb_field_patterns<Fields...>(),
+                                    value != nullptr ? std::string{value->name()} : std::string{}, true);
```
[the remark this answers is not in the document]
Comment (hhenson) [reply include/hgraph/types/type_pattern.h]: Fixed in 7b760a98a. An unresolved NominalTSB now retains its NominalBundle origin and argument ScalarPattern, matches that pattern against the concrete TSB value schema, binds type arguments, and resolves the exact specialization. Regression coverage matches Box<Int>, resolves it, and rejects Other<Int>.

File: language/src/codegen/cpp_emitter.cpp [hunk 3]
```diff
@@ -2386,6 +3122,71 @@ namespace hgl::codegen
...
+            std::vector<std::string> template_parameters;
+            std::vector<std::string> type_arguments;
+            for (std::size_t index = 0; index < item.generics.size(); ++index)
+            {
+                const ast::GenericParameter &parameter = item.generics[index];
+                const std::string name = cpp_name(parameter.name.text);
+                if (parameter.is_const)
+                {
+                    template_parameters.push_back("std::int64_t " + name);
+                    continue;
```
[the remark this answers is not in the document]
Comment (hhenson) [reply language/src/codegen/cpp_emitter.cpp]: Fixed in 7b760a98a. Native emit-cpp now rejects any const-generic struct declaration until hgraph has typed constant Bundle metadata. The regression covers an unused const-generic declaration so it cannot silently collapse specialization identity.
